deployer/src/deployment/state_change_layer.rs

```diff
@@ -187,60 +61,36 @@ where
         }
 
         let mut visitor = NewStateVisitor::default();
-
         attrs.record(&mut visitor);
 
-        let details = visitor.details;
-
-        if details.id.is_nil() {
+        if visitor.id.is_nil() {
             warn!("scope details does not have a valid id");
             return;
         }
 
-        // Safe to unwrap since this is the `on_new_span` method
-        let span = ctx.span(id).unwrap();
-        let mut extensions = span.extensions_mut();
-        let metadata = span.metadata();
-
-        self.recorder.record(Log {
-            id: details.id,
-            state: details.state,
-            level: metadata.level().into(),
-            timestamp: Utc::now(),
-            file: metadata.file().map(str::to_string),
-            line: metadata.line(),
-            target: metadata.target().to_string(),
-            fields: Default::default(),
-            r#type: LogType::State,
+        // To deployer persistence
+        self.state_recorder.record_state(DeploymentState {
+            id: visitor.id,
+            state: visitor.state,
         });
-
-        extensions.insert::<ScopeDetails>(details);
-    }
-}
-
-/// Used to keep track of the current state a deployment scope is in
-#[derive(Debug, Default)]
-struct ScopeDetails {
-    id: Uuid,
-    state: State,
-}
-
-impl From<&tracing::Level> for LogLevel {
-    fn from(level: &tracing::Level) -> Self {
-        match *level {
-            tracing::Level::TRACE => Self::Trace,
-            tracing::Level::DEBUG => Self::Debug,
-            tracing::Level::INFO => Self::Info,
-            tracing::Level::WARN => Self::Warn,
-            tracing::Level::ERROR => Self::Error,
-        }
+        // To logger
+        self.log_recorder.record(LogItem::new(
+            visitor.id,
+            Backend::Deployer,
+            format!(
+                "{} {}",
+                tracing::Level::INFO.colored(),
+                format!("Entering {} state", visitor.state), // make blue?
+            ),
+        ));
```

Comment:
If we register both layers for deployer, then this won't be needed.

Reply:
Not sure I'm following but let's keep it and remove anything that is redundant.

Reply:
Oh, I see. This is the same as the old state change line :+1: